sql-plugin/src/main/scala/org/apache/spark/sql/rapids/GpuFileFormatWriter.scala

```diff
@@ -128,19 +126,17 @@ object GpuFileFormatWriter extends Logging {
 
     val empty2NullPlan = if (needConvert) GpuProjectExec(projectList, plan) else plan
 
-    val bucketIdExpression = bucketSpec.map { _ =>
-      // Use `HashPartitioning.partitionIdExpression` as our bucket id expression, so that we can
-      // guarantee the data distribution is same between shuffle and bucketed data source, which
-      // enables us to only shuffle one side when join a bucketed table and a normal one.
-      //HashPartitioning(bucketColumns, spec.numBuckets).partitionIdExpression
-      //
-      // TODO: Cannot support this until we either:
-      // Guarantee join and bucketing are both on the GPU and disable GPU-writing if join not on GPU
-      //   OR
-      // Guarantee GPU hash partitioning is 100% compatible with CPU hashing
+    val writerBucketSpec = bucketSpec.map { spec =>
+      // TODO: Cannot support this until we:
+      // support Hive hash partitioning on the GPU
       throw new UnsupportedOperationException("GPU hash partitioning for bucketed data is not "
-          + "compatible with the CPU version")
+        + "compatible with the CPU version")
+
+      // add this to let the compiler infer
+      // the type of writerBucketSpec as `Option[GpuWriterBucketSpec]`
+      GpuWriterBucketSpec(null, null)
```

Comment:
This should be getting a dead code warning from the compiler.  If we want the compiler to know that `writerBucketSpec` is of type `GpuWriterBucketSpec` then a much better way is to simply declare it as such, i.e.:
```scala
val writerBucketSpec: GpuWriterBucketSpec = bucketSpec.map { spec =>
```